### PR TITLE
Implement Release view mutations

### PR DESCRIPTION
**Pull request by mozbhearsum**
There's a few other small fixes/changes here too:
* Ensure `useAction` always passes along all positional arguments.
* Disable `Delete` button on `ReleaseCard` instead of hiding it when a Release has Rules pointing at it.
* Replace the `Delete Release` button on the `Release` view with `Cancel Pending Change` (like we have for Rules).

### CODE REVIEW

File: src/hooks/useAction.js [hunk 1]
```diff
@@ -17,12 +17,12 @@ export default action => {
   const [data, setData] = useState(null);
   const [error, setError] = useState(null);
   const state = { loading, data, error };
-  const performAction = async body => {
+  const performAction = async (...body) => {
```
Comment (helfi92) [inline src/hooks/useAction.js]: 🥇

File: src/views/Releases/Release/index.jsx [hunk 2]
```diff
@@ -152,11 +246,11 @@ export default function Release(props) {
             {!isNewRelease && (
```
Comment (helfi92) [inline src/views/Releases/Release/index.jsx]: Similar to the rule view, let's now show the speed dial if the buttons are no changes to cancel.

File: src/views/Releases/Release/index.jsx [hunk 3]
```diff
@@ -90,10 +130,64 @@ export default function Release(props) {
...
+    }
+  };
+
+  const handleReleaseUpdate = async () => {
+    // Updates to Releases require signoff in some circumstances,
+    // but we don't have a need to schedule them for specific times.
+    // Because of this, we always schedule them for slightly in the future,
+    // which means that changes that don't require signoff will happen
+    // almost immediately, and changes that do require signoff will wait
+    // until those are completed.
```
Comment (helfi92) [inline src/views/Releases/Release/index.jsx]: The comments have been really helpful :)

Comment (mozbhearsum) [reply src/views/Releases/Release/index.jsx]: Thanks! I hope my future self thinks so too :).

**Comment by helfi92** [comment]
I think https://github.com/mozilla-frontend-infra/balrog-ui/pull/171#discussion_r316395641 still needs to be done.

File: src/views/Releases/Release/index.jsx (hunk 2 above)
Comment (mozbhearsum) [reply src/views/Releases/Release/index.jsx]: Oh, whoops. I forgot to push before merging, heh.

**Pull request merged**